src/main/java/org/sonarlint/intellij/core/ServerUpdateTask.java

```diff
@@ -107,15 +114,16 @@ private void updateModules(ServerConfiguration serverConfiguration) {
     final Set<String> existingProjectKeys = engine.allModulesByKey().keySet();
     final Set<String> invalidModules = new HashSet<>();
 
-    for (String key : projectKeys) {
-      if (existingProjectKeys.contains(key)) {
-        updateModule(serverConfiguration, key);
+    for (int i = 0; i < projects.size(); i++) {
+      String moduleKey = moduleKeys.get(i);
+      if (existingProjectKeys.contains(moduleKey)) {
+        updateModule(serverConfiguration, projects.get(i), moduleKey);
```

Comment:
As I explained in my previous comment, I'm not sure it's a good idea to assume that a project corresponds to a module key. Anyway the signature of the constructor doesn't make it obvious at all that this would happen.